.github/workflows/backport-action.yml

```diff
@@ -0,0 +1,47 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+---
+name: Automatic backport action
+on:  # yamllint disable-line rule:truthy
+  pull_request_target:
+    types: ['labeled', 'closed']
+    branches: ['main']
+permissions:
+  contents: write
+  pull-requests: write
+jobs:
+  backport:
+    name: Backport PR ${{ github.event.pull_request.number }}
+    if: >
+      github.event.pull_request.merged == true &&
+      contains(join(github.event.pull_request.labels.*.name, ' '), 'backport-to-v')
```

Comment:
Perhaps lets be specific about the versions.

Reply:
I am okay to keep hardcoded variables like this with conditions, 
contains(github.event.pull_request.labels.*.name, 'backport-to-v2-10-test')
contains(github.event.pull_request.labels.*.name, 'backport-to-v2-9-test')
contains(github.event.pull_request.labels.*.name, 'backport-to-v2-8-test')

But which branches do we want to backport to? not sure so just used over branching name structure

Reply:
Yeah. I like the open nature it has. I would even possibly change it to `backport-to-`  -> because maybe we would like to use it to backport to other branches in the future.

Reply:
I wan't so sure how we were getting the branches. This seems good!